src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -35,6 +35,10 @@ struct segment_info_t {
     return state == Segment::segment_state_t::CLOSED;
   }
 
+  bool is_releasing() const {
+    return state == Segment::segment_state_t::RELEASING;
+  }
+
```

Comment:
Is is_releasing only used in assert? if it is only used in an assert, how does it change the code logic to avoid the bug?

Reply:
The behavior change is in get_segment().  segments in this state will return false for is_empty() preventing them from being reused until after commit.

Reply:
so that means some segment doesn't finish release but marked empty?  how can we avoid this?  only one place call segment_cleaner.mark_segment_released (mark_empty) that is in submition_transaction, at this place, submition already finished. why can cause reuse segment in releasing?

Reply:
I don't follow.  The idea is that the segment is in state RELEASING from when we finish scanning the segment until when the last transaction relocating the segment commits, where we call mark_segment_released.  Is the objection that we should be calling mark_segment_released only after the call to segment_manager.release?  That's fair, I'll make that change.

Reply:
Oh, and I got the behavior change backwards.  What I described was already true -- we didn't mark the segment empty until after commit.  The problem was that it remained marked closed until that point which could cause do_gc to start gc'ing it again.

Reply:
See get_next_gc_target.

Reply:
sure, that is the behaver changed not the get_segment(). segments. Got it.